### PR TITLE
document secrets per #100

### PR DESCRIPTION
**Pull request by john-floren-gravwell**

### CODE REVIEW

File: Automation/automation.tex [hunk 1]
```diff
@@ -1060,3 +1060,47 @@ \subsection{Hands-on Lab: Scripting}
...
+
+Secrets are managed via the Secrets page in the Tools \& Resources section of the Gravwell main menu. To create a new secret, click the ``+ Secret'' button in the upper right and fill out the fields in the dialog as shown in Figure \ref{fig:new-secret}. 
+
+\begin{figure}
+	\includegraphics[width=0.5\linewidth]{images/new-secret.png}
+	\caption{Creating a secret}
+	\label{fig:new-secret}
+\end{figure}
+
+Note that once the secret is created, the user will not be able to retrieve the contents of the secret again! It is possible to \emph{change} the contents of the secret, but never to \emph{read} them.
```
Comment (kris-watts-gravwell) [inline Automation/automation.tex:1084]: this is repeated, is that intended?

File: Automation/automation.tex [hunk 2]
```diff
@@ -1060,3 +1060,47 @@ \subsection{Hands-on Lab: Scripting}
...
+
+\subsection{Security Considerations}
+
+Although Gravwell does its best to protect the contents of secrets, there are several fundamental issues which must be faced by *any* system which uses stored secrets in automations. In brief, anyone who has root access to the Gravwell webserver systems can probably extract your secrets with a little work. Further details are given below.
+
+Before creating a secret, think through what could happen if the secret were leaked and how you would respond. Adding a secret containing a Mattermost webhook URL to your internal corporate Gravwell system is low-risk; creating a secret containing your SSH private key on a publicly-accessible Gravwell system is higher-risk.
+
+\begin{itemize}
+\item Secrets are stored unencrypted on the webserver's filesystem (and on the datastore, if used). This means that anyone who can read files in \code{/opt/gravwell/etc} on the Gravwell server(s) can theoretically extract the contents of a secret from the database.
+\item If Gravwell is configured with HTTPS disabled, secrets will cross the network unencrypted; anyone who can sniff network traffic may be able to retrieve them.
```
Comment (kris-watts-gravwell) [inline Automation/automation.tex:1104]: should there be something about "AND the searchagent is not colocated with the webserver" ?

**Review by kris-watts-gravwell — APPROVED** [review]
LGTM if comments don't need to be addressed

File: Automation/automation.tex (hunk 1 above)
Comment (john-floren-gravwell) [reply Automation/automation.tex:1084]: yes

File: Automation/automation.tex (hunk 2 above)
Comment (john-floren-gravwell) [reply Automation/automation.tex:1104]: When you create a secret, the request will go to the webserver unencrypted as well. Basically, if you set up gravwell with http, it's your own fault if somebody sniffs ya.

File: Automation/automation.tex [hunk 3]
```diff
@@ -1060,3 +1060,47 @@ \subsection{Hands-on Lab: Scripting}
...
+
+\begin{figure}
+	\includegraphics[width=0.5\linewidth]{images/secret-sharing.png}
+	\caption{Sharing a secret}
+	\label{fig:secret-sharing}
+\end{figure}
+
+\subsection{Security Considerations}
+
+Although Gravwell does its best to protect the contents of secrets, there are several fundamental issues which must be faced by *any* system which uses stored secrets in automations. In brief, anyone who has root access to the Gravwell webserver systems can probably extract your secrets with a little work. Further details are given below.
```
Comment (ashnwade) [inline Automation/automation.tex:1098]: should this be \emph{any} instead of *any*?

Comment (john-floren-gravwell) [reply Automation/automation.tex:1098]: yes, fixing

**Comment by john-floren-gravwell** [comment]
ptal

**Review by ashnwade — APPROVED** [review]
LGTM

**Pull request merged**